tests/test_max_stable_set.py

```diff
@@ -0,0 +1,582 @@
+                                                                  'routes': {'service_1_route_1', 'service_1_route_2'},
+                                                                  'id': 'stop_1.link:artificial_link===from:stop_1===to:stop_1',
+                                                                  'x': 1.0, 'y': 2.5, 'epsg': 'epsg:27700', 'name': '',
+                                                                  'lon': -7.557148552832129, 'lat': 49.76683027967191,
+                                                                  's2_id': 5205973754090340691,
+                                                                  'additional_attributes': set(),
+                                                                  'linkRefId': 'artificial_link===from:stop_1===to:stop_1',
+                                                                  'stop_id': 'stop_1'}}
+    mss.artificial_links = {
+        'artificial_link===from:stop_1===to:stop_1': {'from': 'stop_1', 'to': 'stop_1', 'modes': {'bus'}},
+        'artificial_link===from:node_6===to:stop_1': {'from': 'node_6', 'to': 'stop_1', 'modes': {'bus'}},
+        'artificial_link===from:stop_1===to:node_5': {'from': 'stop_1', 'to': 'node_5', 'modes': {'bus'}}}
+    mss.pt_edges = DataFrame(
+        {'services': {0: {'bus_service'}, 1: {'bus_service'}, 2: {'bus_service'}, 3: {'bus_service'}},
+         'routes': {0: {'service_1_route_2'}, 1: {'service_1_route_2'}, 2: {'service_1_route_1'},
+                    3: {'service_1_route_1'}},
+         'u': {0: 'stop_3', 1: 'stop_2', 2: 'stop_2', 3: 'stop_1'},
+         'v': {0: 'stop_2', 1: 'stop_1', 2: 'stop_3', 3: 'stop_2'},
+         'key': {0: 0, 1: 0, 2: 0, 3: 0},
+         'linkRefId_u': {0: 'link_7_8_car', 1: 'link_5_6_car', 2: 'link_5_6_car',
+                         3: 'artificial_link===from:stop_1===to:stop_1'},
+         'linkRefId_v': {0: 'link_5_6_car', 1: 'artificial_link===from:stop_1===to:stop_1', 2: 'link_7_8_car',
+                         3: 'link_5_6_car'},
+         'shortest_path': {0: ['link_7_8_car', 'link_8_7_car', 'link_7_6_car', 'link_6_5_car', 'link_5_6_car'],
+                           1: ['link_5_6_car', 'artificial_link===from:node_6===to:stop_1',
+                               'artificial_link===from:stop_1===to:stop_1'],
+                           2: ['link_5_6_car', 'link_6_7_car', 'link_7_8_car'],
+                           3: ['artificial_link===from:stop_1===to:stop_1', 'artificial_link===from:stop_1===to:node_5',
+                               'link_5_6_car']}})
+    mss.unsolved_stops = {'stop_1'}
+    return mss
+
+
+def test_generating_changeset_from_partial_mss_problem(partial_mss):
+    changeset = partial_mss.to_changeset(
+        DataFrame({'ordered_stops': {'service_1_route_2': ['stop_3', 'stop_2', 'stop_1'],
+                                     'service_1_route_1': ['stop_1', 'stop_2', 'stop_3']}})
+    )
+    assert_semantically_equal(
```

Comment:
Is there an argument for each of these assertions becoming its own test? Ideally we want to minimise the number of things that can make a test fail, and this test seems to have a high number of different things that could cause failure.